### DISCUSSION TITLE
Node's diskpools empty after reboot

### DISCUSSION THREAD

**Discussion by mattlqx** [openebs/openebs#3808] [Q&A] [answered]
I have 3 nodes each with several diskpools utilizing `/dev/disk/by-id` device naming. OpenEBS is configured with Mayastor replicated volumes each with 2 replicas.

I performed a graceful shutdown of a node by draining it via `kubectl drain` and then shut it down. When I booted the node back up and returned it to service, all its diskpools (there were 4) then reported as 0 used.  I didn't check into the volume activity while the host was down, but when I looked at `kubectl mayastor get volume-replica-topologies`, I did see all volumes on the two remaining hosts having all configured volumes.

Is this expected behavior? I was thinking the replica on the down host would just get rebuilt from the data it already had, but I suspect Mayastor had already moved replicas to one of the other up hosts and then just deletes the out-of-date volumes when the host comes back up.

**Comment by tiagolobocastro** [comment, marked as answer]
Yeah I'd expect the data would be rebuild from other nodes, and then when the node is booted up again its replicas would just be deleted as they're no longer needed.
